src/creature.h

```diff
@@ -345,39 +356,31 @@ class Creature : virtual public Thing
 	bool registerCreatureEvent(const std::string& name);
 	bool unregisterCreatureEvent(const std::string& name);
 
-	bool hasParent() const override { return getParent(); }
-	Cylinder* getParent() const override final { return tile; }
-	void setParent(Cylinder* cylinder) override final
+	bool hasParent() const override { return getParent() != nullptr; }
+	std::shared_ptr<Thing> getParent() const override final { return tile; }
+	void setParent(std::shared_ptr<Thing> thing) override final
 	{
-		tile = static_cast<Tile*>(cylinder);
+		tile = thing->getTile();
 		position = tile->getPosition();
 	}
 
-	const Position& getPosition() const override final { return position; }
+	const Position& getPosition() const { return position; }
```

Comment:
The override is missing here
```suggestion
	const Position& getPosition() const override { return position; }
```